subsys/net/ip/net_if.c

```diff
@@ -428,7 +431,12 @@ static inline void init_iface(struct net_if *iface)
 #if defined(CONFIG_NET_NATIVE_IPV6)
 	net_if_flag_set(iface, NET_IF_IPV6);
 #endif
-
+#if defined(CONFIG_NET_DSA)
```

Comment:
We have been trying to limit adding driver specific configuration to generic code. So in this respect, this is not the proper place. We could have init function in `subsys/net/l2/ethernet/dsa/dsa.c` and which could be called by `ethernet_init()` in `subsys/net/l2/ethernet/ethernet.c` . That `dsa_init()` could set these things if needed.

Reply:
okay, thanks for suggestion.
Let me try to do that:)